docs/contents/build-stages.md

```diff
@@ -0,0 +1,52 @@
+# Controller Build Stages
```

Comment:
this doc is not as comprehensive as your PR overview - do we have a single place where this process is laid out? Why do we separate into multiple docs instead of a single doc with sections?